pywbem_mock/_baseprovider.py

```diff
@@ -0,0 +1,479 @@
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+A CIM provider creates WBEM server responses to operations defined in DSP0200.
+
+The BaseProvider WBEM server provider handles provider required data that
+is common to both the MainProvider, DefaultInstanceProvider, and any
+registered instance providers including:
+
+  * Access to the CIM repository
+  * Access to the Method repository
+  * Local provider methods that are common to both the main provider and other
+    providers.
+  * CIMInstance request operations except for selected operations that are
+    handled by the BaseInstanceProvider to allow for specialized
+    instance providers to be implemented.
+  * Associator request operations
+
+This provider uses a separate CIM repository for the objects maintained by
+the CIM repository and defined with the interfaces in `BaseRepository`.
+"""
+
+from __future__ import absolute_import, print_function
+
+from copy import deepcopy
+
+from pywbem import CIMError, \
+    CIM_ERR_NOT_FOUND, \
+    CIM_ERR_ALREADY_EXISTS, CIM_ERR_INVALID_NAMESPACE, \
+    CIM_ERR_NAMESPACE_NOT_EMPTY
+
+from pywbem._utils import _format
+
+# pywbem_mock implementation configuration variables that are used in
+# request responsders.
+
+# Default Max_Object_Count for Fake Server if not specified by request
+_DEFAULT_MAX_OBJECT_COUNT = 100
+
+# Maximum Open... timeout if not set by request
+OPEN_MAX_TIMEOUT = 40
+
+# Per DSP0200, the default behavior for EnumerateInstance DeepInheritance
+# if not set by server.  Default is True.
+DEFAULT_DEEP_INHERITANCE = True
+
+# Value of LocalOnly parameter for instance retrevial operations.  This is
+# set to False in this implementation because of issues between different
+# versions of the DSP0200 specification that defined incompatible behavior
+# for this parameter that were resolved in Version 1.4 by
+# stating that False was the recommended Server setting for all instance
+# retrevial requests and that clients should use False also to avoid the
+# incompatibility.
+INSTANCE_RETRIEVE_LOCAL_ONLY = False
+
+
+class BaseProvider(object):
+    """
+    BaseProvider it the top level class in the provider hiearchy and includes
+    the attributes and local methods required by the subclasses such as
+    MainProvider and DefaultInstanceProvider.
+    """
+
+    def __init__(self, conn=None, cimrepository=None):
+        """
+        Parameters:
+            conn(:class:`~pywbem_mock.FakedWBEMConnection`):
+              The current instance of the connection from which some
+              variables are extracted for the repository including
+              the host and disable_pull_operations attributes.
+
+            cimrepository(derived from `~pywbem_mock.BaseRepository`):
+             the CIM repository that stores CIM objects.
+        """
+
+        #   Save required attributes from the connection object
+        if conn:
+            self.host = conn.host
```

Comment:
Instead of passing in `conn` and then using exactly one attribute from it, I suggest to pass in that one attribute. That avoids the back reference and makes it more clear what information is actually passed.

Reply:
**DONE**